zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/publishing/Publisher.java

```diff
@@ -787,7 +789,15 @@ private static void sendToKafka(Collection collection) throws IOException {
     // Valid CMDDataset uris for published CMD versions of a dataset (edition) - /dataset/{datatsetId}/editions/{edition}/versions/{version}/metadata
     protected static boolean isValidCMDDatasetURI (String uri){
 
-        return uri.chars().filter(ch -> ch == '/').count() > 6;
+        int matches = 0;
+        Matcher matcher = Pattern.compile("/").matcher(uri);
+        while(matcher.find()) {
+            matches++;
+        }
+        if (matches>5)
+            return true;
+
+        return false;
```

Comment:
This is suffers from the same problem - the check is not specific enough to accurately identify a CMD dataset uri. For example if the input `uri` value is `/1/2/3/4/5` then that will be a match even though its not a valid CMD version uri.

You're correct in that the original code is a more concise implementation of this, but I was trying to say we should use a more accurate method to match the uris  which is why I suggested a regex - sorry I should have been more clear.

I suggested a regex because a valid CMD version uri has a specific pattern we can match against:

`/datasets/{dataset-id}/editions/{edition}/versions/{version}`

If we use a regex to match this pattern we can be way more accurate in our validation. I'm not sure this example is 100% correct & may need some refining but something to this effect should be what we need:

`^\/datasets\/.+\/editions\/.+\/versions\/[0-9]+$`

If we compile this as a regex we should be able to more accurately match values. I use this [online tool](https://www.regexpal.com/) to help me build up the regex I need in isolation which is pretty useful as it means you don't need to run the code to test it.

Reply:
Done, new regex pattern applied, pls let me know the specific expected regex, if not, which meets the requirements.